hbase-common/src/main/java/org/apache/hadoop/hbase/util/FutureUtils.java

```diff
@@ -132,9 +165,7 @@ public static Throwable unwrapCompletionException(Throwable error) {
     } catch (InterruptedException e) {
       throw (IOException) new InterruptedIOException().initCause(e);
     } catch (ExecutionException e) {
-      Throwable cause = e.getCause();
-      Throwables.propagateIfPossible(cause, IOException.class);
-      throw new IOException(cause);
+      throw rethrow(e);
```

Comment:
rethrow(e)?

Reply:
There will be a compile error if you do not have a return or throw statement in this block...